etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/RuleEngine.java

```diff
@@ -0,0 +1,60 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+
+/** Manages the application of rules loaded from a definitions file using the RuleLoader. */
+public class RuleEngine {
+
+    private final Path RULES_DEFINITIONS_PATH =
+            Path.of("../etor/src/main/resources/rule_definitions.json");
+    private boolean rulesLoaded = false;
+
+    private static final RuleEngine INSTANCE = new RuleEngine();
+
+    final List<Rule> rules = new ArrayList<>();
+
+    @Inject Logger logger;
+    @Inject RuleLoader ruleLoader;
+
+    private RuleEngine() {}
+
+    public static RuleEngine getInstance() {
+        return INSTANCE;
+    }
+
+    public void unloadRules() {
+        rulesLoaded = false;
+        rules.clear();
+    }
+
+    public void ensureRulesLoaded() {
+        if (!rulesLoaded) {
+            logger.logDebug("Loading rules definitions from " + RULES_DEFINITIONS_PATH);
+            try {
+                var loadedRules = ruleLoader.loadRules(RULES_DEFINITIONS_PATH);
+                rules.addAll(loadedRules);
+                rulesLoaded = true;
+            } catch (RuleLoaderException e) {
+                logger.logError(
+                        "Failed to load rules definitions from: " + RULES_DEFINITIONS_PATH, e);
+            }
+        }
+    }
+
+    public void validate(IBaseResource resource) {
+        logger.logDebug("Validating FHIR resource");
+        ensureRulesLoaded();
```

Comment:
Should this be inside of the constructor on line 24 to load the rules when we start the application?

Reply:
This can be done, but I have a concern in that the method needs the logger to be loaded, and that makes the injects need to be written above the construction.

Reply:
I have a [similar comment below](https://github.com/CDCgov/trusted-intermediary/pull/943#discussion_r1535713313) with another possible solution.

Reply:
Added a potential fix on the related thread.